src/main/scala/nl/vroste/zio/kinesis/client/AdminClientLive.scala

```diff
@@ -0,0 +1,300 @@
+package nl.vroste.zio.kinesis.client
+
+import java.time.Instant
+
+import nl.vroste.zio.kinesis.client.Util.{ asZIO, paginatedRequest }
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient
+import software.amazon.awssdk.services.kinesis.model._
+import zio.clock.Clock
+import zio.duration._
+import zio.interop.reactivestreams._
+import zio.stream.ZStream
+import zio.{ Tag => _, _ }
+
+import scala.jdk.CollectionConverters._
+
+object AdminClientLive {
```

Comment:
Can we make this a `private class AdminClientLive extends AdminClient.Service` with a constructor parameter `kinesisClient` and in `object AdminClient` have a `val live: ZLayer[.., AdminClient]`? 

I think `AdminClient.live` is a bit more user-friendly than `AdminClientLive.layer`, and I think that's how the module pattern is commonly applied in zio libs.

Reply:
Nice idea re private class, I'll give it a go

Reply:
done